gdx/src/com/badlogic/gdx/utils/LongArray.java

```diff
@@ -272,11 +272,13 @@ public boolean removeAll (LongArray array) {
 
 	/** Removes and returns the last item. */
 	public long pop () {
+		if (size == 0) throw new IllegalStateException("Array is empty.");
```

Comment:
Good catch here; calling pop() on an empty LongArray would produce a negative `size` and only then throw an Exception. Catching that would still mean dealing with negative size later!

Reply:
One possible nitpick, now that I think about it, is that if the IllegalStateException is caught once, and pop() is called again, it will throw an IndexOutOfBoundsException just after the catch, but won't throw an IllegalStateException a second time, because size will be -1, and that won't be == 0. We might prefer size <= 0 because the size can be set externally, plus this method can make the size negative, and all sizes 0 or less are invalid here.